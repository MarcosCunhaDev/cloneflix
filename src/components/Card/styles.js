import styled from "styled-components/native";
import { colors } from "../../config/const";

export const BoxCard = styled.SafeAreaView`
  width: 120px;
  height: 160px;
  justify-content: flex-start;
  align-items: center;
  margin: 10px 10px;
  overflow: hidden;
  border-radius: 5px;
`;

export const ContentImage = styled.View`
  width: 120px;
  height: 130px;
  border-radius: 5px;
  overflow: hidden;
`;
export const BoxTitle = styled.View`
  border-radius: 5px;
  padding: 5px;
  padding-left: 0px;
  padding-horizontal: 6px;
  height: 30px;
  width: 100%;
  flex-direction: row;
  align-items: center;
`;
export const Title = styled.Text`
  color: white;
`;
