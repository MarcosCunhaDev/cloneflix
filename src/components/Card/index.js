import React from "react";
import { View, Text, Image } from "react-native";
import { BoxCard, Title, ContentImage, BoxTitle } from "./styles";

const Card = ({ title, uri }) => {
  return (
    <BoxCard>
      <ContentImage>
        <Image
          source={{
            uri:
              uri ||
              "https://upload.wikimedia.org/wikipedia/pt/f/f5/Marvels_avengers_capa.jpg",
          }}
          style={{ flex: 1 }}
        />
      </ContentImage>
      <BoxTitle>
        <Title numberOfLines={1} maxLength={5}>
          {title}
        </Title>
      </BoxTitle>
    </BoxCard>
  );
};

export default Card;
