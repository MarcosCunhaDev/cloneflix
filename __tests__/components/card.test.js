import React from "react";
import renderer from "react-test-renderer";

import Card from "../../src/components/Card";

describe("<Card />", () => {
  it("has 2 children", () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});

it("renders correctly", () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
