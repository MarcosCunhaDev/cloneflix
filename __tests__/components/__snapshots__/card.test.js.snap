// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "height": 160,
        "justifyContent": "flex-start",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 10,
        "marginTop": 10,
        "overflow": "hidden",
        "width": 120,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 5,
          "height": 130,
          "overflow": "hidden",
          "width": 120,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://image.tmdb.org/t/p/originalundefined",
        }
      }
      style={
        Object {
          "flex": 1,
          "resizeMode": "cover",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 30,
          "paddingBottom": 5,
          "paddingHorizontal": 6,
          "paddingLeft": 0,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      maxLength={5}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "white",
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
